### PR TITLE
Fix scroll to zoom jump between 25% and 1000%

### LINKED ISSUE #6013: Ctrl + scroll to zoom jump from 25% to 1000%
With continuous scroll enabled mouse (magic mouse), <kbd>Ctrl</kbd> + <kbd>scroll down</kbd> change zoom level to 1000% and <kbd>Ctrl</kbd> + <kbd>scroll up</kbd> change zoom level to 25% regardless the current zoom level.

OS X 10.10.3 (14D131)
Chrome 42.0.2311.135 (64-bit)
Safari 8.0.5 (10600.5.17)

### PR DESCRIPTION
**Pull request by xlc**
Fixes #6013

### CODE REVIEW

**Comment by Snuffleupagus** [comment]
I'm sorry, but this patch seem to "break" normal scroll wheels, since the loops in `zoomIn/zoomOut` now runs one *more* time than it should.

**Comment by Snuffleupagus** [comment]
I'm sorry, but this patch seem to "break" normal scroll wheels, since the loop in `zoomIn/zoomOut` now runs one *more* time than it should.
I'm assuming that the actual issue this patch tries to address is when `zoomIn/zoomOut` is called with the argument `0`.

File: web/viewer.js [hunk 1]
```diff
@@ -291,7 +291,8 @@ var PDFViewerApplication = {
       newScale = (newScale * DEFAULT_SCALE_DELTA).toFixed(2);
       newScale = Math.ceil(newScale * 10) / 10;
       newScale = Math.min(MAX_SCALE, newScale);
-    } while (--ticks && newScale < MAX_SCALE);
+      --ticks;
+    } while (ticks >= 0 && newScale < MAX_SCALE);
```
Comment (Snuffleupagus) [inline web/viewer.js]: I think that changing these lines to `} while (--ticks > 0 && newScale < MAX_SCALE);` should achieve what you want here (similar below).

**Comment by xlc** [comment]
@Snuffleupagus Fixed. The actual issue is the tick count is not always integer, so `--ticks` will reduce it to negative value.

**Comment by Snuffleupagus** [comment]
Thanks for the patch!

**Pull request merged**